ERCS/erc-7730.md

```diff
@@ -0,0 +1,1033 @@
+
+When specified a wallet MUST only use specified sources to resolve address names. Wallet MUST verify the type of the address if able to. 
+
+When omitted, a wallet MAY use any source to resolve an address.
+
+| Address type  | Description |
+| ---           | --- |
+| wallet        | Address is an account controlled by the wallet |
+| eoa           | Address is an Externally Owned Account         |
+| contract      | Address is a well known smartcontract  |
+| token         | Address is a well known ERC-20 token  |
+| collection    | Address is a well known NFT collection  |
+
+A wallet MAY verify that a `wallet` address is in fact controlled by the wallet, and reject signing if not the case.
+
+| Source type   | Description |
+| ---           | --- |
+| local         | Address MAY be replaced with a local name trusted by user. Wallets MAY consider that `local` setting for `sources` is always valid |
+| ens           | Address MAY be replaced with an associated ENS domain |
+
+### Wallets
+
+#### Ledger
+
+[TBD] Link to ledger specification
+
+
+## Rationale
+
+### Human readability
+
+It is expected that the main limitation to adoption of ERC-7730 will be the complexity of writing this interface description file compared to interest of writing it.
+
+This drove a few choices when introducing this ERC specification:
+* The form of an ERC itself will allow usage of these file by any wallets (Hardware or Software, without restrictions), and in turn drive up the benefits for dApps developers to provide their own ERC-7730 description
+* The specification is intended to be directly readable by developers, in order to facilitate direct edition by developers.
+* In addition, a set of edition tools will be created and open sourced to ease visualization of the results for end users
+
+### Wallet limitations
+
+Wide support by wallets is key for adoption of this specification. 
+
+Hardware wallets tend to have more limited capabilities that will impact what they can display securely, especially since the intention of the specification is to create a strong security binding between the spec and the data being reviewed.
+
+This consideration is driving a few choices done for ERC-7730:
+* Complex UI constructs like layouts, grouping and re-ordering of fields have been left to a wallet specific section, yet unspecified. After a time, we may see patterns emerge between wallets in term of minimal features.
+* Representation of fields has been created to allow "flattening" the list of fields, when handling complex message structures. This flattened representation is expected to work better with Hardware wallets in particular, and is recommended at first.
+* Some formatters that require recursive constructs, like `calldata` are expected to work with restrictions at first, especially on Hardware wallets.  
+
+## Test Cases
+
+More examples can be found on the Ledger ERC-7730 public repository
```

Comment:
This is an external link. Please remove it, or move the content into your `assets/` directory.